include/vicon/udp.h

```diff
@@ -228,28 +242,52 @@ class ObjectReference
 template<typename ObjectDataType = ObjectData>
 class UDPClient
 {
-public:
-    UDPClient() {}
+private:
+    /*
+     * NB: The first member of this struct *should* be object ID, but the
+     * Vicon system seems to just give a value of zero for every item, which
+     * isn't terribly helpful. So instead we distinguish objects based on their
+     * names.
+     */
```

Comment:
If this is the case - which sucks - it might be better to switch the whole thing to using a ``std::map``/``std::unordered_map`` - my original design only really makes sense if the numeric ids do :disappointed:

Reply:
That's probably a good idea